File: CS50_classes/CS50x/readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    float L = (float) (count_letters(text)) / count_words(text) * 100;
    float S = (float) (count_sentences(text)) / count_words(text) * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int count_letters(string text)
{
    int counter = 0;

    for (int i = 0; text[i] != '\0'; i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            counter++;
        }
    }

    return counter;
}

int count_words(string text)
{
    int counter = 1;

    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == 32)
        {
            counter++;
        }
    }

    return counter;
}

int count_sentences(string text)
{
    int counter = 0;

    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            counter++;
        }
    }

    return counter;
}

File: CS50x/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    L = count_letters(text) / count_words(text) * 100
    S = count_sentences(text) / count_words(text) * 100

    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade", index)


def count_letters(text):
    counter = 0

    for c in text:
        if "A" <= c <= "Z" or "a" <= c <= "z":
            counter += 1

    return counter


def count_words(text):
    counter = 1

    for c in text:
        if c == " ":
            counter += 1

    return counter


def count_sentences(text):
    counter = 0

    for c in text:
        if c in [".", "!", "?"]:
            counter += 1

    return counter


if __name__ == "__main__":
    main()
